### PR TITLE
Pay orders with pba

### PR DESCRIPTION
**Pull request by vigneshpushkaranHMCTS**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: model/src/main/java/uk/gov/hmcts/payment/api/jpaaudit/listner/PaymentFeeLinkEntityListener.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    @Transactional(MANDATORY)
+    private void perform(PaymentFeeLink target, Action action, AuditEventsType auditEventsType) {
+        EntityManager entityManager = BeanUtil.getBean(EntityManager.class);
+
+//        String ccdCaseNumber =target.getCaseDetails().iterator().next().getCcdCaseNumber();
+//        Will be used once all services onboarded orders endpoint
+        entityManager.persist(PaymentAuditHistory.paymentAuditHistoryWith()
+            .ccdCaseNo(target.getCcdCaseNumber())
```
Comment (AyanDuttaHMCTS) [inline model/src/main/java/uk/gov/hmcts/payment/api/jpaaudit/listner/PaymentFeeLinkEntityListener.java:32]: if target.getCcdCaseNumber() would work here ? I think Order should retrieve the ccd_case from CASE_DETAILS through ORDER_CASE

**Review by AyanDuttaHMCTS — CHANGES_REQUESTED** [review]
Vignesh Can you please have a look into the review comments

File: model/src/main/java/uk/gov/hmcts/payment/api/dto/FeePayApportionCCDCase.java [hunk 2]
```diff
@@ -1,5 +1,6 @@
 package uk.gov.hmcts.payment.api.dto;
 
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
 import com.fasterxml.jackson.databind.PropertyNamingStrategy;
```
[the remark this answers is not in the document]
Comment (vigneshpushkaranHMCTS) [reply model/src/main/java/uk/gov/hmcts/payment/api/dto/FeePayApportionCCDCase.java]: Unused are removed now

File: model/src/main/java/uk/gov/hmcts/payment/api/dto/PciPalPaymentRequest.java [hunk 3]
```diff
@@ -1,6 +1,7 @@
 package uk.gov.hmcts.payment.api.dto;
 
 import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
 import com.fasterxml.jackson.databind.PropertyNamingStrategy;
```
[the remark this answers is not in the document]
Comment (vigneshpushkaranHMCTS) [reply model/src/main/java/uk/gov/hmcts/payment/api/dto/PciPalPaymentRequest.java]: Unused are removed now

File: model/src/main/java/uk/gov/hmcts/payment/api/dto/PciPalPayment.java [hunk 4]
```diff
@@ -2,6 +2,7 @@
 
 
 import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
 import com.fasterxml.jackson.databind.PropertyNamingStrategy;
```
[the remark this answers is not in the document]
Comment (vigneshpushkaranHMCTS) [reply model/src/main/java/uk/gov/hmcts/payment/api/dto/PciPalPayment.java]: Unused are removed now

File: model/src/main/java/uk/gov/hmcts/payment/api/jpaaudit/listner/PaymentFeeLinkEntityListener.java (hunk 1 above)
Comment (vigneshpushkaranHMCTS) [reply model/src/main/java/uk/gov/hmcts/payment/api/jpaaudit/listner/PaymentFeeLinkEntityListener.java:32]: I commented the line 29,30 which will be used once other controllers store data in the caseDetails table.
//        String ccdCaseNumber =target.getCaseDetails().iterator().next().getCcdCaseNumber();
//        Will be used once all services onboarded orders endpoint

File: gov-pay-client/src/main/java/uk/gov/hmcts/payment/api/external/client/dto/CreatePaymentRequest.java [hunk 5]
```diff
@@ -3,6 +3,7 @@
 package uk.gov.hmcts.payment.api.external.client.dto;
 
 import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
 import com.fasterxml.jackson.databind.PropertyNamingStrategy;
```
[the remark this answers is not in the document]
Comment (vigneshpushkaranHMCTS) [reply gov-pay-client/src/main/java/uk/gov/hmcts/payment/api/external/client/dto/CreatePaymentRequest.java]: Unused are removed now

File: gov-pay-client/src/main/java/uk/gov/hmcts/payment/api/external/client/dto/RefundPaymentRequest.java [hunk 6]
```diff
@@ -3,6 +3,7 @@
 package uk.gov.hmcts.payment.api.external.client.dto;
 
 import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
 import com.fasterxml.jackson.databind.PropertyNamingStrategy;
```
[the remark this answers is not in the document]
Comment (vigneshpushkaranHMCTS) [reply gov-pay-client/src/main/java/uk/gov/hmcts/payment/api/external/client/dto/RefundPaymentRequest.java]: Unused are removed now

File: gov-pay-client/src/main/java/uk/gov/hmcts/payment/api/external/client/dto/RefundSummary.java [hunk 7]
```diff
@@ -2,14 +2,15 @@
 package uk.gov.hmcts.payment.api.external.client.dto;
 
 import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
 import com.fasterxml.jackson.databind.PropertyNamingStrategy;
```
[the remark this answers is not in the document]
Comment (vigneshpushkaranHMCTS) [reply gov-pay-client/src/main/java/uk/gov/hmcts/payment/api/external/client/dto/RefundSummary.java]: Unused are removed now

File: gov-pay-client/src/main/java/uk/gov/hmcts/payment/api/external/client/dto/TelephonyProviderAuthorisationResponse.java [hunk 8]
```diff
@@ -2,6 +2,7 @@
 
 import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
 import com.fasterxml.jackson.databind.PropertyNamingStrategy;
```
[the remark this answers is not in the document]
Comment (vigneshpushkaranHMCTS) [reply gov-pay-client/src/main/java/uk/gov/hmcts/payment/api/external/client/dto/TelephonyProviderAuthorisationResponse.java]: Unused are removed now

File: gov-pay-client/src/main/java/uk/gov/hmcts/payment/api/external/client/dto/TelephonyProviderLinkIdResponse.java [hunk 9]
```diff
@@ -2,6 +2,7 @@
 
 import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
 import com.fasterxml.jackson.databind.PropertyNamingStrategy;
```
[the remark this answers is not in the document]
Comment (vigneshpushkaranHMCTS) [reply gov-pay-client/src/main/java/uk/gov/hmcts/payment/api/external/client/dto/TelephonyProviderLinkIdResponse.java]: Unused are removed now

File: model/build.gradle [hunk 10]
```diff
@@ -7,7 +7,7 @@ dependencies {
     compile group: 'org.springframework.boot', name: 'spring-boot-starter-web'
     compile group: 'org.springframework.boot', name: 'spring-boot-starter-data-jpa'
     compile group: 'org.springframework.security.oauth', name: 'spring-security-oauth2', version: '2.5.0.RELEASE'
-
+    compile group: 'uk.gov.hmcts.reform', name: 'service-auth-provider-client', version: '3.1.1'
```
[the remark this answers is not in the document]
Comment (vigneshpushkaranHMCTS) [reply model/build.gradle:10]: Yes, We are having dependency of authToken generator in  ReferenceDataService to call refdata endpoint.

File: api-contract/src/main/java/uk/gov/hmcts/payment/api/contract/CreditAccountPaymentRequest.java [hunk 11]
```diff
@@ -2,25 +2,25 @@
 
 import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
 import com.fasterxml.jackson.databind.PropertyNamingStrategy;
```
[the remark this answers is not in the document]
Comment (vigneshpushkaranHMCTS) [reply api-contract/src/main/java/uk/gov/hmcts/payment/api/contract/CreditAccountPaymentRequest.java]: Unused are removed now

File: api-contract/src/main/java/uk/gov/hmcts/payment/api/contract/FeeDto.java [hunk 12]
```diff
@@ -1,6 +1,7 @@
 package uk.gov.hmcts.payment.api.contract;
 
 import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
 import com.fasterxml.jackson.databind.PropertyNamingStrategy;
```
[the remark this answers is not in the document]
Comment (vigneshpushkaranHMCTS) [reply api-contract/src/main/java/uk/gov/hmcts/payment/api/contract/FeeDto.java]: Unused are removed now

File: api-contract/src/main/java/uk/gov/hmcts/payment/api/contract/PaymentAllocationDto.java [hunk 13]
```diff
@@ -2,6 +2,7 @@
 
 import com.fasterxml.jackson.annotation.JsonFormat;
 import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
 import com.fasterxml.jackson.databind.PropertyNamingStrategy;
```
[the remark this answers is not in the document]
Comment (vigneshpushkaranHMCTS) [reply api-contract/src/main/java/uk/gov/hmcts/payment/api/contract/PaymentAllocationDto.java]: Unused are removed now

File: api-contract/src/main/java/uk/gov/hmcts/payment/api/contract/PaymentsResponse.java [hunk 14]
```diff
@@ -1,5 +1,6 @@
 package uk.gov.hmcts.payment.api.contract;
 
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
 import com.fasterxml.jackson.databind.PropertyNamingStrategy;
```
[the remark this answers is not in the document]
Comment (vigneshpushkaranHMCTS) [reply api-contract/src/main/java/uk/gov/hmcts/payment/api/contract/PaymentsResponse.java]: Unused are removed now

File: api-contract/src/main/java/uk/gov/hmcts/payment/api/contract/StatusHistoryDto.java [hunk 15]
```diff
@@ -2,6 +2,7 @@
 
 import com.fasterxml.jackson.annotation.JsonFormat;
 import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
 import com.fasterxml.jackson.databind.PropertyNamingStrategy;
```
[the remark this answers is not in the document]
Comment (vigneshpushkaranHMCTS) [reply api-contract/src/main/java/uk/gov/hmcts/payment/api/contract/StatusHistoryDto.java]: Unused are removed now

File: api-contract/src/main/java/uk/gov/hmcts/payment/api/contract/TelephonyCardPaymentsResponse.java [hunk 16]
```diff
@@ -3,6 +3,7 @@
 import com.fasterxml.jackson.annotation.JsonFormat;
 import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
 import com.fasterxml.jackson.databind.PropertyNamingStrategy;
```
[the remark this answers is not in the document]
Comment (vigneshpushkaranHMCTS) [reply api-contract/src/main/java/uk/gov/hmcts/payment/api/contract/TelephonyCardPaymentsResponse.java]: Unused are removed now

File: api-contract/src/main/java/uk/gov/hmcts/payment/api/contract/UpdatePaymentRequest.java [hunk 17]
```diff
@@ -1,6 +1,7 @@
 package uk.gov.hmcts.payment.api.contract;
 
 import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
 import com.fasterxml.jackson.databind.PropertyNamingStrategy;
```
[the remark this answers is not in the document]
Comment (vigneshpushkaranHMCTS) [reply api-contract/src/main/java/uk/gov/hmcts/payment/api/contract/UpdatePaymentRequest.java]: Unused are removed now

File: api/src/main/java/uk/gov/hmcts/payment/api/domain/model/OrderFeeBo.java [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+    private String code;
+
+    private String version;
+
+    private Integer volume;
+
+    private BigDecimal calculatedAmount;
+
+    private String ccdCaseNumber; // Will be removed after get api's work without ccd dependency
+
```
[the remark this answers is not in the document]
Comment (vigneshpushkaranHMCTS) [reply api/src/main/java/uk/gov/hmcts/payment/api/domain/model/OrderFeeBo.java:32]: Yes, its don'e in OrderDto request body.

File: api-contract/src/main/java/uk/gov/hmcts/payment/api/contract/BulkScanningReportDto.java [hunk 19]
```diff
@@ -1,5 +1,6 @@
 package uk.gov.hmcts.payment.api.contract;
 
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
 import com.fasterxml.jackson.databind.PropertyNamingStrategy;
```
[the remark this answers is not in the document]
Comment (vigneshpushkaranHMCTS) [reply api-contract/src/main/java/uk/gov/hmcts/payment/api/contract/BulkScanningReportDto.java]: Unused are removed now

File: api-contract/src/main/java/uk/gov/hmcts/payment/api/contract/BulkScanningUnderOverPaymentDto.java [hunk 20]
```diff
@@ -1,5 +1,6 @@
 package uk.gov.hmcts.payment.api.contract;
 
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
 import com.fasterxml.jackson.databind.PropertyNamingStrategy;
```
[the remark this answers is not in the document]
Comment (vigneshpushkaranHMCTS) [reply api-contract/src/main/java/uk/gov/hmcts/payment/api/contract/BulkScanningUnderOverPaymentDto.java]: Unused are removed now

File: api-contract/src/main/java/uk/gov/hmcts/payment/api/contract/CardPaymentRequest.java [hunk 21]
```diff
@@ -2,17 +2,18 @@
 
 import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
 import com.fasterxml.jackson.databind.PropertyNamingStrategy;
```
[the remark this answers is not in the document]
Comment (vigneshpushkaranHMCTS) [reply api-contract/src/main/java/uk/gov/hmcts/payment/api/contract/CardPaymentRequest.java]: Unused are removed now

File: api/src/main/java/uk/gov/hmcts/payment/api/domain/model/OrderFeeBo.java (hunk 18 above)
[the remark this answers is not in the document]
Comment (AyanDuttaHMCTS) [reply api/src/main/java/uk/gov/hmcts/payment/api/domain/model/OrderFeeBo.java:32]: Right, its fine then

File: api/src/main/java/uk/gov/hmcts/payment/api/domain/service/OrderDomainService.java [hunk 22]
```diff
@@ -0,0 +1,25 @@
...
+
+    PaymentFeeLink find(String orderReference);
+
+    String create (OrderDto orderDto, MultiValueMap<String, String> headers);
+
+    void addFees (OrderBo orderBo, List<OrderFeeDto> feeDtos);
+
+    OrderPaymentBo addPayments (PaymentFeeLink order, OrderPaymentDto orderPaymentDto) throws CheckDigitException;
+
+    Boolean isDuplicate(String orderReference);
```
[the remark this answers is not in the document]
Comment (AyanDuttaHMCTS) [reply api/src/main/java/uk/gov/hmcts/payment/api/domain/service/OrderDomainService.java:18]: park this for integration with other API

**Review by AyanDuttaHMCTS — APPROVED** [review]

**Pull request closed without merging**